File: app/src/main/java/funnybrain/hsu/idv/taipeipark/ParkAdapter.java
package funnybrain.hsu.idv.taipeipark;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import funnybrain.hsu.idv.taipeipark.restful.model.Park;

public class ParkAdapter extends RecyclerView.Adapter<ParkAdapter.ViewHolder> {
    private final Context mContext;
    private List<Park> list;

    public ParkAdapter(Context context, List<Park> list) {
        mContext = context;
        this.list = list;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_park, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.title.setText(list.get(position).getParkName() + "\n" + list.get(position).getPicName());
        Glide.with(mContext).load(list.get(position).getImage()).into(holder.image);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.iv_img) ImageView image;
        @BindView(R.id.tv_title) TextView title;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/funnybrain/hsu/idv/taipeipark/ParkAdapter.kt
package funnybrain.hsu.idv.taipeipark

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import funnybrain.hsu.idv.taipeipark.restful.model.Park
import kotlinx.android.synthetic.main.item_park.view.*
import org.jetbrains.anko.layoutInflater

class ParkAdapter(private val list: List<Park>) : RecyclerView.Adapter<ParkAdapter.ViewHolder>() {

    lateinit var ctx: Context

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        ctx = parent.context
        val view = parent.context.layoutInflater.inflate(R.layout.item_park, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.title.text = " ${list[position].parkName} \n ${list[position].picName}"
        Glide.with(ctx).load(list[position].image).into(holder.image)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val image: ImageView = view.iv_img
        val title: TextView = view.tv_title
    }
}
